core/src/main/java/nu/yona/server/util/AsyncExecutor.java

```diff
@@ -38,10 +41,47 @@ public ThreadData getThreadData()
 		return new ThreadData(MDC.getCopyOfContextMap(), headersHolder.export());
 	}
 
-	public void initThreadAndDo(ThreadData threadData, Runnable action)
+	@Async
+	public void execAsync(ThreadData threadData, Runnable action, Consumer<Optional<Throwable>> completionHandler)
 	{
-		threadData.contextMap.ifPresent(MDC::setContextMap);
-		headersHolder.importFrom(threadData.headers);
-		action.run();
+		try (Handler handler = Handler.initialize(headersHolder, threadData))
+		{
+			try
+			{
+				action.run();
+				completionHandler.accept(Optional.empty());
+			}
+			catch (Throwable e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1181)

Reply:
Here we really want to catch Throwable